kong/tools/database_cache.lua

```diff
@@ -157,4 +174,30 @@ function _M.get_or_set(key, cb)
   return value
 end
 
+function _M.get_or_set(key, cb)
+    return get_or_set(key, cb, nil, nil)
+end
+
+function _M.rawget_or_rawset(key, cb, exptime)
+    return get_or_set(key, cb, exptime, true)
+end
+
+function _M.get_or_init(key, value, exptime)
+    local success, err = _M.add(key, value, exptime)
+    if success then
+        return value
+    else
+        return _M.get(key)
+    end
+end
+
+function _M.rawget_or_init(key, value, exptime)
+    local success, err = _M.rawadd(key, value, exptime)
+    if success then
+        return value
+    else
+        return _M.rawget(key)
+    end
+end
+
```

Comment:
Could we remove all changes from this file except the `rawadd` addition? This module is legacy and bound to disappear and we certainly don't want it to get over-bloated by unused code.